charts/opentelemetry-operator/templates/admission-webhooks/operator-webhook.yaml

```diff
@@ -0,0 +1,182 @@
+{{- if and (.Values.admissionWebhooks.enabled) (not .Values.admissionWebhooks.certManager.enabled) }}
+{{- $altNames := list ( printf "%s.%s" "opentelemetry-operator-webhook-service" .Release.Namespace ) ( printf "%s.%s.svc" "opentelemetry-operator-webhook-service" .Release.Namespace ) -}}
+{{- $ca := genCA "opentelemetry-operator-operator-ca" 365 -}}
+{{- $cert := genSignedCert (include "opentelemetry-operator.fullname" .) nil $altNames 365 $ca -}}
+apiVersion: v1
+kind: Secret
```

Comment:
Is there a way to break this out into its own thing so the objects below don't have to be duplicated?

Reply:
make sense, I will simplify this

Reply:
I tried like this:
```yaml
{{- if .Values.admissionWebhooks.enabled }}
{{- if not .Values.admissionWebhooks.certManager.enabled }}
{{- $altNames := list ( printf "%s.%s" "opentelemetry-operator-webhook-service" .Release.Namespace ) ( printf "%s.%s.svc" "opentelemetry-operator-webhook-service" .Release.Namespace ) -}}
{{- $ca := genCA "opentelemetry-operator-operator-ca" 365 -}}
{{- $cert := genSignedCert (include "opentelemetry-operator.fullname" .) nil $altNames 365 $ca -}}
apiVersion: v1
kind: Secret
type: kubernetes.io/tls
metadata:
  annotations:
    "helm.sh/hook": "pre-install,pre-upgrade"
    "helm.sh/hook-delete-policy": "before-hook-creation"
  labels:
    {{- include "opentelemetry-operator.labels" . | nindent 4 }}
  name: opentelemetry-operator-controller-manager-service-cert
  namespace: {{ .Release.Namespace }}
data:
  tls.crt: {{ $cert.Cert | b64enc }}
  tls.key: {{ $cert.Key | b64enc }}
---
{{- end }}
apiVersion: admissionregistration.k8s.io/v1
kind: MutatingWebhookConfiguration
metadata:
{{- if .Values.admissionWebhooks.certManager.enabled }}
  annotations:
    cert-manager.io/inject-ca-from: {{ printf "%s/%s-serving-cert" .Release.Namespace (include "opentelemetry-operator.name" .) }}
{{- end }}
  labels:
    app.kubernetes.io/name: opentelemetry-operator
  name: {{ template "opentelemetry-operator.name" . }}-mutating-webhook-configuration
webhooks:
  - admissionReviewVersions:
      - v1
    clientConfig:
{{- if not .Values.admissionWebhooks.certManager.enabled }}
      caBundle: {{ $ca.Cert | b64enc }}       👈 # parse error at (opentelemetry-operator/templates/admission-webhooks/operator-webhook.yaml:37): undefined variable "$ca"
{{- end }}
```

But It failed to parse variable "$ca":
 https://github.com/open-telemetry/opentelemetry-helm-charts/actions/runs/3060941363/jobs/4940081825#step:4:27

And I followed `tailingsidecar` project: 
- https://github.com/SumoLogic/tailing-sidecar/blob/main/helm/tailing-sidecar-operator/templates/_operator-webhook-with-cert-manager.tpl
- https://github.com/SumoLogic/tailing-sidecar/blob/main/helm/tailing-sidecar-operator/templates/_operator-webhook.tpl

they used seprated file to impl this 🤔

Reply:
Thanks for sharing that example.  I see now why it is done this way,